src/test/java/org/opensearch/security/dlic/dlsfls/StarTreeDlsFlsTest.java

```diff
@@ -0,0 +1,453 @@
+        HttpResponse response = rh.executePostRequest(
+            "/" + STARTREE_INDEX_NAME + "/_search?pretty",
+            aggregationQuery,
+            encodeBasicHeader("startree_admin", "password")
+        );
+        assertThat(response.getStatusCode(), is(HttpStatus.SC_OK));
+
+        /**
+         * "aggregations" : {
+         *     "departments" : {
+         *       "doc_count_error_upper_bound" : 0,
+         *       "sum_other_doc_count" : 0,
+         *       "buckets" : [
+         *         {
+         *           "key" : "engineering",
+         *           "doc_count" : 2,
+         *           "total_sales" : {
+         *             "value" : 2500.0
+         *           }
+         *         },
+         *         {
+         *           "key" : "sales",
+         *           "doc_count" : 2,
+         *           "total_sales" : {
+         *             "value" : 4500.0
+         *           }
+         *         }
+         *       ]
+         *     }
+         *   }
+         */
+        Assert.assertTrue(
+            "Should contain correct sales total",
+            response.getBody().contains("2500.0") && response.getBody().contains("4500.0")
+        );
+
+        // Check star tree stats - should show star tree queries were used
+        HttpResponse statsResponse = rh.executeGetRequest(
+            "/" + STARTREE_INDEX_NAME + "/_stats/search?pretty",
+            encodeBasicHeader("startree_admin", "password")
+        );
+        assertThat(statsResponse.getStatusCode(), is(HttpStatus.SC_OK));
+
+        Assert.assertTrue(
+            "Star tree query total should be 1 since there are no restrictions",
+            statsResponse.getBody().contains("\"startree_query_total\" : 1")
+        );
+
+        Assert.assertTrue(
+            "Regular query total should be > 0",
+            statsResponse.getBody().contains("\"query_total\" : 1")
+                || statsResponse.getBody().matches(".*\"query_total\"\\s*:\\s*[1-9]\\d*.*")
+        );
+
+    }
+
+    @Test
+    public void testStarTreeWithDlsRestriction() throws Exception {
+        setupStarTreeTest();
+
+        String aggregationQuery = """
+            {
+              "size": 0,
+              "aggs": {
+                "departments": {
+                  "terms": {
+                    "field": "department"
+                  },
+                  "aggs": {
+                    "total_sales": {
+                      "sum": {
+                        "field": "sales_amount"
+                      }
+                    }
+                  }
+                }
+              }
+            }
+            """;
+
+        HttpResponse response = rh.executePostRequest(
+            "/" + STARTREE_INDEX_NAME + "/_search?pretty",
+            aggregationQuery,
+            encodeBasicHeader("startree_dls_user", "password")
+        );
+        assertThat(response.getStatusCode(), is(HttpStatus.SC_OK));
```

Comment:
can you add the aggregation response assertion. Explicitly what numbers are expected?